### PR TITLE
fix(FOROME-682): fixed dtree stat fetching

### PR DESCRIPTION
**Pull request by AlMaQntr**

### CODE REVIEW

**Comment by AndreyUstyumenko** [comment]
need to discuss

File: src/service-providers/decision-trees/decision-trees.adapters.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import { IDtreeStatResponse } from './decision-trees.interface'
+import { TDtreeStat } from './decision-trees.types'
+
+export const adoptDtreeStatResponse = (
```
Comment (AndreyUstyumenko) [inline src/service-providers/decision-trees/decision-trees.adapters.ts]: typo **adapt**

File: src/service-providers/decision-trees/decision-trees.types.ts [hunk 2]
```diff
@@ -0,0 +1,7 @@
+import { TItemsCount, TPropertyStatus } from '../common'
```
Comment (AndreyUstyumenko) [inline src/service-providers/decision-trees/decision-trees.types.ts]: move types to decision-trees.interface.ts

File: src/service-providers/decision-trees/decision-trees.provider.ts [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+type TGetFullDtreeStatParams = {
+  ds: string
+  no: string
+  code: string
+}
+
+type TGetFullDtreeStatOptions = {
+  abortSignal?: AbortSignal
+  onPartialResponse?: (response: TDtreeStat) => void
+}
```
Comment (AndreyUstyumenko) [inline src/service-providers/decision-trees/decision-trees.provider.ts]: move types to decision-trees.interface.ts

File: src/service-providers/decision-trees/decision-trees.types.ts (hunk 2 above)
Comment (AlMaQntr) [reply src/service-providers/decision-trees/decision-trees.types.ts]: done

File: src/service-providers/decision-trees/decision-trees.provider.ts (hunk 3 above)
Comment (AlMaQntr) [reply src/service-providers/decision-trees/decision-trees.provider.ts]: done

File: src/service-providers/decision-trees/decision-trees.adapters.ts (hunk 1 above)
Comment (AlMaQntr) [reply src/service-providers/decision-trees/decision-trees.adapters.ts]: done

**Review by AndreyUstyumenko — APPROVED** [review]

**Review by Lapk1n — APPROVED** [review]

**Review by ghost — APPROVED** [review]

**Pull request merged**